catalog/resource_connection.go

```diff
@@ -46,10 +22,28 @@ func suppressPemPrivateKeyExpiration(k, old, new string, d *schema.ResourceData)
 }
 
 func ResourceConnection() common.Resource {
-	s := common.StructToSchema(ConnectionInfo{},
+	s := common.StructToSchema(catalog.ConnectionInfo{},
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			m["name"].DiffSuppressFunc = common.EqualFoldDiffSuppress
 			m["options"].DiffSuppressFunc = suppressPemPrivateKeyExpiration
+			common.CustomizeSchemaPath(m, "url").SetReadOnly()
+			common.CustomizeSchemaPath(m, "metastore_id").SetReadOnly()
+			common.CustomizeSchemaPath(m, "credential_type").SetReadOnly()
+			common.CustomizeSchemaPath(m, "connection_id").SetReadOnly()
+			common.CustomizeSchemaPath(m, "created_at").SetReadOnly()
+			common.CustomizeSchemaPath(m, "created_by").SetReadOnly()
+			common.CustomizeSchemaPath(m, "full_name").SetReadOnly()
+			common.CustomizeSchemaPath(m, "provisioning_info").SetReadOnly()
+			common.CustomizeSchemaPath(m, "securable_type").SetReadOnly()
+			common.CustomizeSchemaPath(m, "updated_at").SetReadOnly()
+			common.CustomizeSchemaPath(m, "updated_by").SetReadOnly()
```

Comment:
minor nit, should we turn this into a for loop?

Reply:
yes 🤦